uqcsbot/scripts/uq_course_util.py

```diff
@@ -0,0 +1,176 @@
+from uqcsbot import bot
+import requests
+from datetime import datetime
+from dateutil import parser
+from bs4 import BeautifulSoup
+from functools import partial
+
+BASE_COURSE_URL = 'https://my.uq.edu.au/programs-courses/course.html?course_code='
+BASE_ASSESSMENT_URL = 'https://www.courses.uq.edu.au/student_section_report.php?report=assessment&profileIds='
+BASE_CALENDAR_URL = 'http://www.uq.edu.au/events/calendar_view.php?category_id=16&year='
+
+class DateSyntaxException(Exception):
+    '''
+    Raised when an unparsable date syntax is encountered.
+    '''
+    def __init__(self, date):
+        super().__init__()
+        self.date = date
+        bot.logger.error(f'Could not parse date \'{date}\'.')
```

Comment:
I really don't like the `bot.logger.error` calls in the constructor - you can replace them with setting the message in the superclass call

Reply:
An example of a way I'd prefer it:
```python
def __init__(self, date):
    self.message = f'Could not parse date \'{date}\'.'
    self.date = date
    super().__init__(self.message, self.date)
```
and then maybe also
```python
def __str__(self):
    return self.message
```